crates/swc_ecma_lints/src/rules/no_console.rs

```diff
@@ -0,0 +1,31 @@
+use crate::rule::{visitor_rule, Rule};
+use swc_common::errors::HANDLER;
+use swc_ecma_ast::*;
+use swc_ecma_visit::{noop_visit_type, Visit};
+
+const MESSAGE: &'static str = "Unexpected console statement";
+
+pub fn no_console() -> Box<dyn Rule> {
+    visitor_rule(NoConsole::default())
+}
+
+#[derive(Debug, Default)]
+struct NoConsole;
+
+impl NoConsole {
+    fn check(&mut self, id: &Ident) {
+        if id.sym.eq("console") {
+            if id.span.ctxt.outer().as_u32() == 1 {
```

Comment:
This is wrong.

Reply:
You have to check if it's `top_level_ctxt`, which should be provided on creation.

Reply:
I can't understand where I should put top_level_ctxt into `fn no_console()`. It means that I should call `no_console(some_top_level_ctx)` ?

Reply:
Yes

Reply:
then I should compare top file context and identifier span context?

smth like?

```
if id.span == self.top_file_context {
  // write warn
}
```

Reply:
Not span itself,`.span.ctxt`

Reply:
Ok. And one more question. How to get access to file top level context? Should I use visit_module/visit script for it?

Reply:
> You have to check if it's top_level_ctxt, which should be provided on creation

You should accept it from `no_console`.

Reply:
It's correct? 

```
let m: Module = ....;
// do stuff

let rule = no_console(m.span.ctxt.clone());
rule.lint_module(&m);
```

Reply:
No. I mean `pub fn no_console(top_level_ctxt: SyntaxContext)`

Reply:
Above I assume rule call code.

Of course I'll add argument to no_console fn, place SyntaxContext to struct and other things

Reply:
No, the span of the module does not have any context.
You should use the `Mark` (or `SytnaxContext`) passed to `resolver_with_mark`.

Reply:
Can you get my any example? A bit hard to understand

Reply:
Many passes accept `top_level_mark` or `top_level_ctxt`. See https://github.com/swc-project/swc/blob/a4e93eb2dff0614e7a39c4301c84654412cf68dc/crates/swc/src/config/mod.rs#L367-L456

Top level context here is `SyntaxContext::empty().apply_mark(top_level_mark)`.